src/gluonts/shell/sagemaker/params.py

```diff
@@ -79,3 +78,95 @@ def encode_sagemaker_parameters(decoded_params: dict) -> dict:
     {'foo': '[1, 2, 3]', 'bar': 'hello'}
     """
     return map_dct_values(encode_sagemaker_parameter, decoded_params)
+
+
+def detrim_and_decode_sagemaker_parameters(trimmed_params: dict) -> dict:
+    """Decode a SageMaker parameters dictionary where all values are strings.
+
+    Example:
+
+    >>> detrim_sagemaker_parameters({
+    ...     '_0_foo': '[1, ', '_1_foo': '2, 3', '_2_foo': ']',
+    ...      '_0_bar': 'hell', '_1_bar': 'o'
+    ... })
+    {'foo': [1, 2, 3], 'bar': 'hello'}
+    """
+    encoded_params = detrim_sagemaker_parameters(trimmed_params)
+    return map_dct_values(decode_sagemaker_parameter, encoded_params)
+
+
+def encode_and_trim_sagemaker_parameters(
+    decoded_params: dict, max_len: int = 256
+) -> dict:
+    """Encode a SageMaker parameters dictionary where all values are strings then
+    trim them to account for Sagemaker character size limit.
+
+    Example:
+
+    >>> encode_and_trim_sagemaker_parameters({
+    ...     "foo": [1, 2, 3],
+    ...     "bar": "hello"
+    ... }, max_len = 4)
+    {
+        '_0_foo': '[1, ', '_1_foo': '2, 3', '_2_foo': ']',
+        '_0_bar': 'hell', '_1_bar': 'o'
+    }
+    """
+    endoded_params = map_dct_values(encode_sagemaker_parameter, decoded_params)
+    return trim_encoded_sagemaker_parameters(endoded_params, max_len)
+
+
+def trim_encoded_sagemaker_parameters(
+    encoded_params: dict, max_len: int = 256
+) -> dict:
+    """Trim parameters that have already been encoded to a given max length.
+
+    Example:
+    >>> trim_encoded_sagemaker_parameters({
+    ...     'foo': '[1, 2, 3]',
+    ...     'bar': 'hello'
+    ... }, max_len = 4)
+    {
+        '_0_foo': '[1, ', '_1_foo': '2, 3', '_2_foo': ']',
+        '_0_bar': 'hell', '_1_bar': 'o'
+    }
+    """
+    trimmed_params = {}
+    for key, value in encoded_params.items():
+        if len(value) > max_len:
+            for idx, sub_len in enumerate(range(0, len(value), max_len)):
```

Comment:
There is `gluonts.dataset.util.batcher`, which would make this a bit more readable in my opinion:

```python
for idx, substr in enumerate(batcher(value, max_len)):
    trimmed_params[f"_{idx}_{key}"] = "".join(substr)
```

Reply:
added! ideally it would have been nice to remove the  `if len(value) > max_len` using the batcher too, but then all values with appropriate length would have a key prefixed with `_0_`.